### PR TITLE
Feature fix bug of not effective version of mysql56 #203

### PR DESCRIPTION
**Pull request by hustjieke**
### Which issue(s) this PR fixes?

Fixes #

### What this PR does?

Summary:

### CODE REVIEW

**Comment by hustjieke** [comment]
> Resolve conflicts.

Resolved.

File: mysqlcluster/mysqlcluster.go [hunk 1]
```diff
@@ -44,6 +46,8 @@ const (
 	gb
 )
 
+var log = logf.Log.WithName("cluster")
```
Comment (runkecheng) [inline mysqlcluster/mysqlcluster.go:49]: mysqlcluster is better?

File: mysqlcluster/mysqlcluster.go [hunk 2]
```diff
@@ -114,12 +118,19 @@ func (c *MysqlCluster) GetSelectorLabels() labels.Set {
...
+	// 8.0 --> 8.0.x
+	if v, ok := utils.MySQLTagsToSemVer[c.Spec.MysqlVersion]; ok {
 		version = v
+	} else {
+		errmsg := "Invalid mysql version option:" + c.Spec.MysqlVersion
+		log.Error(errors.New(errmsg), "currently we do not support mysql 5.6 or earlier version, default mysql version option should be 5.7 or 8.0")
+		return utils.InvalidMySQLVersion
 	}
 
+	// check if has specified image
```
Comment (runkecheng) [inline mysqlcluster/mysqlcluster.go:133]: The first letter needs capitalization and the end of the sentence needs to be added `.`

**Review by runkecheng — CHANGES_REQUESTED** [review]

File: mysqlcluster/mysqlcluster.go [hunk 3]
```diff
@@ -114,12 +118,19 @@ func (c *MysqlCluster) GetSelectorLabels() labels.Set {
 
 // GetMySQLVersion returns the MySQL server version.
 func (c *MysqlCluster) GetMySQLVersion() string {
-	version := c.Spec.MysqlVersion
-	// lookup for an alias, usually this will solve 5.7 to 5.7.x
-	if v, ok := utils.MySQLTagsToSemVer[version]; ok {
+	var version string
+	// lookup for an alias, this will solve MySQL tags:
+	// 5.7 --> 5.7.x
+	// 8.0 --> 8.0.x
```
Comment (andyli029) [inline mysqlcluster/mysqlcluster.go:124]: Delete the line. When support 8.0, add the comment.

File: mysqlcluster/mysqlcluster.go (hunk 2 above)
Comment (hustjieke) [reply mysqlcluster/mysqlcluster.go]: Done.

File: mysqlcluster/mysqlcluster.go (hunk 3 above)
Comment (hustjieke) [reply mysqlcluster/mysqlcluster.go]: Done.

File: mysqlcluster/mysqlcluster.go (hunk 1 above)
Comment (hustjieke) [reply mysqlcluster/mysqlcluster.go]: Good idea!

**Review by runkecheng — APPROVED** [review]

**Review by andyli029 — COMMENTED** [review]
LGTM.

**Pull request merged**